src/Compilers/CSharp/Portable/Symbols/TypeWithAnnotations.cs

```diff
@@ -489,26 +489,19 @@ internal TypeWithAnnotations SubstituteTypeCore(AbstractTypeMap typeMap)
             {
                 newAnnotation = NullableAnnotation;
             }
-            else if (NullableAnnotation != NullableAnnotation.Oblivious)
+            else if (NullableAnnotation.IsNotAnnotated())
             {
-                if (!typeSymbol.IsTypeParameterDisallowingAnnotationInCSharp8())
```

Comment:
>IsTypeParameterDisallowingAnnotationInCSharp8 [](start = 32, length = 45)

Where else do we use this helper and whether we still need to use it, given that we now allow annotations on type parameters like that?

Reply:
Unless it's crucial for this discussion or PR, I'd rather not branch out to that question.

---
In reply to: [506777623](https://github.com/dotnet/roslyn/pull/48694#discussion_r506777623) [](ancestors = 506777623)